hot/patch.js

```diff
@@ -8,6 +8,4 @@
 /**
  * Apply React patches in development by importing `haul/hot/patch`.
  */
-if (module.hot && process.env.NODE_ENV !== 'production') {
-  require('../src/hot/client/hotPatch');
-}
+require('react-hot-loader/patch');
```

Comment:
As @greg-dove said, we need to disable proxy creation, otherwise HMR won't persist the state:
```js
require('react-hot-loader/patch').default.disableProxyCreation = true;
```